### ISSUE TITLE
conda search broken for me on master

### ISSUE THREAD

**Issue by ilanschnell** [conda/conda#2531]
On master (`4d7286cddf091c0bcf8bd105ad847dd9f57c1ed7`), I now get:
```
$ conda search
...
Traceback (most recent call last):
  File "/Users/ilan/python/bin/conda", line 6, in <module>
    sys.exit(conda.cli.main())
  File "/Users/ilan/conda/conda/cli/main.py", line 120, in main
    args_func(args, p)
  File "/Users/ilan/conda/conda/cli/main.py", line 127, in args_func
    args.func(args, p)
  File "/Users/ilan/conda/conda/cli/main_search.py", line 118, in execute
    execute_search(args, parser)
  File "/Users/ilan/conda/conda/cli/main_search.py", line 155, in execute_search
    extracted = conda.install.extracted()
  File "/Users/ilan/conda/conda/install.py", line 749, in extracted
    return set(dist for dist, rec in package_cache().items() if rec['dirs'])
  File "/Users/ilan/conda/conda/install.py", line 673, in package_cache
    add_cached_package(pdir, fn)
  File "/Users/ilan/conda/conda/install.py", line 627, in add_cached_package
    url = remove_binstar_tokens(url)
  File "/Users/ilan/conda/conda/config.py", line 258, in remove_binstar_tokens
    return BINSTAR_TOKEN_PAT.sub(r'\1', url)
TypeError: expected string or buffer
```

**Comment by ilanschnell** [comment]
Hmm, I also get this when I do `conda install`.  CC @kalefranz

**Comment by kalefranz** [comment]
Guessing you'll be fine on py2 if you want to keep testing. Will definitely address it though ASAP. No idea how it happened or why tests passed with it.

**Comment by ilanschnell** [comment]
I am on Python 2.

**Comment by kalefranz** [comment]
Oh then try py3

**Comment by ilanschnell** [comment]
I've added a debug print statement to see what `url` is.  The result is that `url` is `None`.  So this is not a Python 2 / 3 problem.

**Comment by mcg1969** [comment]
I think I know what it is

**Comment by mcg1969** [comment]
I introduced this in `mgrant/install-standalone` when attempting to handle constructor installations with no  corresponding URL information. You can reproduce this by removing all references in `pkgs\urls.txt` to any package in your package cache. Fixed in #2354.

**Comment by mcg1969** [comment]
I'm leaving `mgrant/install-standalone` open for awhile longer... it has been good for two PRs, why not a third? :-)
